src/brod_sup.erl

```diff
@@ -0,0 +1,88 @@
+%%%
+%%%   Copyright (c) 2015 Klarna AB
+%%%
+%%%   Licensed under the Apache License, Version 2.0 (the "License");
+%%%   you may not use this file except in compliance with the License.
+%%%   You may obtain a copy of the License at
+%%%
+%%%       http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%%   Unless required by applicable law or agreed to in writing, software
+%%%   distributed under the License is distributed on an "AS IS" BASIS,
+%%%   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%%%   See the License for the specific language governing permissions and
+%%%   limitations under the License.
+%%%
+
+%%%=============================================================================
+%%% @doc
+%%% @copyright 2015 Klarna AB
+%%% @end
+%%%=============================================================================
+-module(brod_sup).
+-behaviour(brod_supervisor).
+
+-export([ start_link/0
+        , start_client/2
+        , init/1
+        ]).
+
+-include("brod_int.hrl").
+
+%%%_* APIs ---------------------------------------------------------------------
+
+%% @doc Start root supervisor.
+%%
+%% To start permanent clients, a minimal example of app env (sys.config):
+%%    [ { clients
+%%      , [ {client_1
+%%          , [ {endpoints, [{"localhost", 9092}]}
+%%            ]
+%%          }
+%%        , {client_2, ...}
+%%        ]
+%%      }
+%%    ].
+%% TODO: add permanent producers
+%% @end
+-spec start_link() -> {ok, pid()}.
+start_link() ->
+  brod_supervisor:start_link({local, ?MODULE}, ?MODULE, clients_sup).
+
+%% @doc Start client under brod_sup.
+-spec start_client(client_id(), client_config()) -> ok | {error, any()}.
+start_client(ClientId, Config) ->
+  case brod_supervisor:find_child(brod_sup, ClientId) of
```

Comment:
Because kafka does not require it. Why force users to invent artificial names just to be able to use the library?